core/src/test/java/org/apache/calcite/test/enumerable/EnumerableCorrelateTest.java

```diff
@@ -86,18 +88,17 @@
         .query(
             "select empid, name from emps e where e.deptno in (select d.deptno from depts d)")
         .withHook(Hook.PLANNER, (Consumer<RelOptPlanner>) planner -> {
-          // force the semi-join to run via EnumerableCorrelate instead of EnumerableJoin/SemiJoin
-          planner.addRule(JoinToCorrelateRule.SEMI);
-          planner.removeRule(EnumerableRules.ENUMERABLE_JOIN_RULE);
+          // force the semi-join to run via EnumerableCorrelate
+          // instead of EnumerableHashJoin/SemiJoin
+          planner.addRule(JoinToCorrelateRule.JOIN);
           planner.removeRule(EnumerableRules.ENUMERABLE_SEMI_JOIN_RULE);
         })
         .explainContains(""
             + "EnumerableCalc(expr#0..2=[{inputs}], empid=[$t0], name=[$t2])\n"
-            + "  EnumerableCorrelate(correlation=[$cor1], joinType=[semi], requiredColumns=[{1}])\n"
+            + "  EnumerableHashJoin(condition=[=($1, $3)], joinType=[semi])\n"
```

Comment:
@vlsi You are right, i have force the test cases to produce `EnumerableCorrelate`, but because we deprecate `JoinToCorrelateRule#SEMI`, now the `Correlate` with an `agg` as one input is a cheaper plan.